stdlib/public/Concurrency/Task.swift

```diff
@@ -379,6 +379,97 @@ struct JobFlags {
   }
 }
 
+// ==== Task Creation Flags --------------------------------------------------
+
+/// Flags for schedulable jobs.
+///
+/// This is a port of the C++ FlagSet.
+@available(SwiftStdlib 5.5, *)
+struct TaskCreateFlags {
+  /// The actual bit representation of these flags.
+  var bits: Int = 0
+
+  /// The priority given to the job.
+  var priority: TaskPriority? {
+    get {
+      let value = Int(bits) & 0xFF
+
+      if value == 0 {
+        return nil
+      }
+
+      return TaskPriority(rawValue: UInt8(value))
+    }
+
+    set {
+      bits = (bits & ~0xFF) | Int(newValue?.rawValue ?? 0)
+    }
+  }
+
+  /// Whether this is a child task.
+  var isChildTask: Bool {
+    get {
+      (bits & (1 << 8)) != 0
+    }
+
+    set {
+      if newValue {
+        bits = bits | 1 << 8
+      } else {
+        bits = (bits & ~(1 << 8))
+      }
+    }
+  }
+
+  /// Whether to copy thread locals from the currently-executing task into the
+  /// newly-created task.
+  var copyThreadLocals: Bool {
+    get {
+      (bits & (1 << 10)) != 0
+    }
+
+    set {
+      if newValue {
+        bits = bits | 1 << 10
+      } else {
+        bits = (bits & ~(1 << 10))
+      }
+    }
+  }
+
+  /// Whether this task should inherit as much context from the
+  /// currently-executing task as it can.
+  var inheritContext: Bool {
+    get {
+      (bits & (1 << 11)) != 0
+    }
+
+    set {
+      if newValue {
+        bits = bits | 1 << 11
+      } else {
+        bits = (bits & ~(1 << 11))
+      }
+    }
+  }
+
+  /// Whether to enqueue the newly-created task on the given executor (if
+  /// specified) or the global executor.
+  var enqueueJob: Bool {
```

Comment:
nice idea :) was quite annoying to deal with the always having to enqueue "on the outside" (e.g. in swift code again)...

Reply:
Yeah, I'm trying to get the Swift entry points down to "set of flags, call into the C++ code".